apps/smart-contracts/core/helpers/index.ts

```diff
@@ -26,3 +34,91 @@ export async function createMarket(marketParams: CreateMarketParams): Promise<Cr
+  collateralAmount: BigNumber
+): Promise<BigNumber> {
+  const factorAfterFee = BigNumber.from(FEE_DENOMINATOR).sub(await collateral.getDepositFee())
+  return collateralAmount.mul(factorAfterFee).div(FEE_DENOMINATOR)
+}
+
+export async function checkDepositCap(
+  recipient: string,
+  baseTokenAmount: BigNumber,
+  collateral: Collateral | MockContract<Collateral>
+): Promise<void> {
+  const depositHookAddress = await collateral.getDepositHook()
+  if (depositHookAddress !== ZERO_ADDRESS) {
+    const depositHook = (await ethers.getContractAt(
+      'DepositHook',
+      await collateral.getDepositHook()
+    )) as DepositHook
+    const depositRecordAddress = await depositHook.getDepositRecord()
+    if (depositRecordAddress !== ZERO_ADDRESS) {
+      const depositRecord = (await ethers.getContractAt(
+        'DepositRecord',
+        await depositHook.getDepositRecord()
+      )) as DepositRecord
+      const userDepositCap = await depositRecord.getUserDepositCap()
+      const currentDepositTotal = await depositRecord.getUserDepositAmount(recipient)
+      const newDepositTotal = currentDepositTotal.add(baseTokenAmount)
+      if (newDepositTotal.gt(userDepositCap)) {
+        throw new Error(
+          `${recipient} new deposit total of ${formatEther(
+            newDepositTotal
+          )} exceeds the cap ${formatEther(userDepositCap)}`
+        )
+      }
+    }
+  }
+}
+
+export async function mintCollateralFromBaseToken(
+  funder: SignerWithAddress,
+  recipient: string,
+  baseTokenAmount: BigNumber,
+  collateral: Collateral | MockContract<Collateral>
+): Promise<BigNumber> {
+  await checkDepositCap(recipient, baseTokenAmount, collateral)
+  const baseToken = (await ethers.getContractAt('ERC20', await collateral.getBaseToken())) as ERC20
+  await baseToken.connect(funder).approve(collateral.address, baseTokenAmount)
+  await collateral.connect(funder).deposit(recipient, baseTokenAmount)
+  const mintEvents = await findTransferEvent(collateral, ZERO_ADDRESS, recipient)
+  return mintEvents[0].args.value
+}
+
+export async function mintLSFromCollateral(
+  funder: SignerWithAddress,
+  collateralAmount: BigNumber,
+  market: PrePOMarket | MockContract<PrePOMarket>
+): Promise<BigNumber> {
+  const collateral = (await ethers.getContractAt(
+    'Collateral',
+    await market.getCollateral()
+  )) as Collateral
+  await collateral.connect(funder).approve(market.address, collateralAmount)
+  await market.connect(funder).mint(collateralAmount)
+  return collateralAmount
+}
+
+export async function mintLSFromBaseToken(
+  funder: SignerWithAddress,
+  recipient: SignerWithAddress,
+  baseTokenAmount: BigNumber,
+  market: PrePOMarket | MockContract<PrePOMarket>
+): Promise<BigNumber> {
+  const collateral = (await ethers.getContractAt(
+    'Collateral',
+    await market.getCollateral()
+  )) as Collateral
+  const baseTokenNeeded = await getBaseTokenAmount(collateral, baseTokenAmount)
```

Comment:
shouldn't `collateralAmount` be passed in rather than `baseTokenAmount` (which wouldn't make sense)?
```suggestion
  const baseTokenAmount = await getBaseTokenAmount(collateral, collateralAmount)
```

Reply:
This would conflict with `baseTokenAmount` that is an existing parameter